lecture3/main.c

```diff
@@ -0,0 +1,193 @@
+#include "task3.c"
```

Comment:
C files should not be included within other C files. It is a very bad practice. Header files were created for that purpose.

It probably compiled for you, because you've only passed main.c to the compiler. But a normal practice is when compiling code, you should pass all the c files to the compiler.

If I do it that way task 3 doesn't compile for me.

Considering that, I'd like you to separate all 4 subtasks of this task 3 into 4 separate modules, each with its own c and h file. In the main.c, header files of these modules should be included